packages/replayio/src/utils/recordings/upload/uploadRecordings.ts

```diff
@@ -30,7 +33,26 @@ export async function uploadRecordings(
 
   const multiPartUpload = await getFeatureFlagValue<boolean>("cli-multipart-upload", false);
   const client = new ProtocolClient();
-  await client.waitUntilAuthenticated();
+  try {
+    await client.waitUntilAuthenticated();
+  } catch (error) {
+    if (
+      error instanceof ProtocolError &&
+      error.protocolCode === AUTHENTICATION_REQUIRED_ERROR_CODE
+    ) {
+      let message = `${statusFailed("✘")} Authentication failed.`;
+      if (process.env.REPLAY_API_KEY || process.env.RECORD_REPLAY_API_KEY) {
+        const name = process.env.REPLAY_API_KEY ? "REPLAY_API_KEY" : "RECORD_REPLAY_API_KEY";
+        message += ` Please check your ${highlight(name)}.`;
+      } else {
+        message += " Please try to `replay login` again.";
+      }
+      console.error(message);
+      await exitProcess(1);
+      return;
+    }
+    throw error;
+  }
```

Comment:
Can you share a screenshot of what this looks like?

Reply:
It isn't pretty but it doesn't step out of the comfort zone of what can already happen today since it's handled by this top-level handler:
https://github.com/replayio/replay-cli/blob/9398b58e6c92000a9a7839c7eaab520c9236023d/packages/replayio/src/index.ts#L18-L26

So it just gets logged to stderr and the process exits. We could decorate this with some "Please raise a ticket to help us fix this issue" but I think it's worth printing the error with the call stack and all. It should help us to diagnose problems when people raise issues about seeing this

Reply:
My bad for the unclear comment. I was asking about a screenshot of the formatted prompt above (` console.error(message)`) just to make sure spacing and colors looked okay.

Reply:
Huh. How would auth fail _after_ you'd made a recording? We require auth before we even launch the browser.

Reply:
Yes, but we never validate the token from the env variable - we only test its presence. We could try to validate it, I guess - but then we'd be postponing the recording where *usually* it's probably fine to continue

Reply:
Oh right. This is the API_KEY scenario.